src/PanelTraits/AutoSet.php

```diff
@@ -165,19 +168,37 @@ public function makeLabel($value)
     /**
      * Get the database column names, in order to figure out what fields/columns to show in the auto-fields-and-columns functionality.
      *
-     * @return [array] Database column names as an array.
+     * @param bool $filtered
+     *
+     * @return array Database column names as an array.
      */
-    public function getDbColumnsNames()
+    public function getDbColumnsNames($filtered = true)
     {
-        // Automatically-set columns should be both in the database, and in the $fillable variable on the Eloquent Model
-        $columns = $this->model->getConnection()->getSchemaBuilder()->getColumnListing($this->model->getTable());
-        $fillable = $this->model->getFillable();
+        if (empty($this->db_column_types)) {
+            $this->getDbColumnTypes();
+        }
 
-        if (! empty($fillable)) {
-            $columns = array_intersect($columns, $fillable);
+        //build the db_column_names array only once
+        if (empty($this->db_column_names)) {
+            // Automatically-set columns should be both in the database, and in the $fillable variable on the Eloquent Model
+            $this->db_column_names = array_keys($this->db_column_types);
+
+            if ( $filtered) {
+                $columns  = $this->db_column_names;
+                $fillable = $this->model->getFillable();
+                if ( ! empty($fillable)) {
+                    $columns = array_intersect($columns, $fillable);
+                }// but not updated_at, deleted_at
+                $this->db_column_names = array_values(array_diff($columns,
+                    [
+                        $this->model->getKeyName(),
+                        $this->model->getCreatedAtColumn(),
```

Comment:
What if they DON'T have these? I think there is a way to check this, shouldn't there?
(and why is the soft deleted column left out?)

Reply:
True. I was just using the existing code. But there doesn't need to be a check because requesting a non-existing model attribute -- ```created_at``` for instance -- returns null, and a null entry has no effect on array_diff.
```deleted_at``` isn't left out, it's at the end of the list. But you're right, it should be ```$this->model->getDeletedAtColumn()``` since it might be something else.

Reply:
Ah yes, the infamous "you mistype a model name and spend six days figuring out why your app is broken" trick.